### PR TITLE
Update CLI Install Section (CLI Revisions 1/2)

### PR DESCRIPTION
**Pull request by jmintuitive**
### Description
This is one half of the CLI docs updates. This updates the CLI installation section, and the CLI configuration steps. 

There will be a follow up PR to update the "CLI Use Cases" section. 

This also fixes 2 spots where "deep links" were broken by a header change in the CLI sections.

This includes:
1. Redirects from old content to new content
2. Updates to the sidebar
3. Content changes for files

### Checklist

- Do all Lints pass?
  - [] Have you ran `pnpm spellcheck`?
  - [X] Have you ran `pnpm fmt`?
  - [X] Have you ran `pnpm lint`?

Note: I was unable to get `pnpm spellcheck` to run on my machine - any advice on how to run it would be helpful! (As of now it just immediately exits with `pnpm run spellcheck exited (1) ELIFECYCLE`

### CODE REVIEW

**Comment by jmintuitive** [comment]
Note: As this is just the 1st half, I made sure the content was slightly modified to use existing links (so this won't point to empty pages while I finish up the 2nd half PR :) )

File: apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-specific-version.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+title: "Install Specific Versions (Advanced)"
+---
+
+# Install Specific Aptos CLI Versions (Advanced)
+
+If you need a specific version of the Aptos CLI, you can build it directly from the Aptos source code. This installation method is primarily used to interact with specific features on Devnet which may not have made it to Testnet / Mainnet yet. You may also want to follow these steps if you are running architecture which does not play well with the existing releases / pre-compiled binaries.
```
Comment (gregnazario) [inline apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-specific-version.md:7]:
```suggestion
If you need a specific version of the Aptos CLI, you can build it directly from the Aptos source code. This installation method is primarily used to interact with specific features on Devnet which may not have made it to Testnet / Mainnet yet. You may also want to follow these steps if you are running an architecture which does not play well with the existing releases / pre-compiled binaries.
```

File: apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-windows.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+title: "Install On Windows"
+---
+
+# Install the Aptos CLI on Windows
+
+For Windows, the easiest way to install the Aptos CLI tool is via Python script, although if that does not work, you can also install manually via downloading pre-compiled binaries. The pre-compiled binaries approach is not generally recommended as updating is very manual.
```
Comment (gregnazario) [inline apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-windows.md:7]: I think this is a run on sentence 
```suggestion
For Windows, the easiest way to install the Aptos CLI tool is via Python script. If that does not work, you can also install manually via pre-compiled binaries. The pre-compiled binaries approach is not generally recommended as updating is very manual.
```

File: apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-mac.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+---
+title: "Install On Mac"
+---
+
+# Install the Aptos CLI on Mac
+
+For mac, the easiest way to install the Aptos CLI is with the package manager `brew`.
```
Comment (gregnazario) [inline apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-mac.md:7]:
```suggestion
For Mac, the easiest way to install the Aptos CLI is with the package manager `brew`.
```

File: apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+Example command for [`oh my zsh`](https://ohmyz.sh/):
+
+```zsh
+aptos config generate-shell-completions --shell zsh --output-file ~/.oh-my-zsh/completions/_aptos
+```
+
+## (Optional) Global Config
+
+By default, the configuration is set to use a local file (`.aptos/config.yaml`) per workspace. If you would like to use one shared configuration for all workspaces, you can follow these steps.
```
Comment (gregnazario) [inline apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md:79]:
```suggestion
By default, the configuration is set to use a local file (`.aptos/config.yaml`) per workspace in the current directory. If you would like to use one shared configuration for all workspaces, you can follow these steps.
```

File: apps/docusaurus/docs/tools/aptos-cli/index.md [hunk 5]
```diff
@@ -6,8 +6,19 @@ hidden: false
 
 # Aptos CLI
 
-The `aptos` tool is a command line interface (CLI) for developing on the Aptos blockchain, debugging, and for node operations. Please refer to the following documentation for more information:
+The Aptos command line interface (CLI) is a tool to help you compile and test Move contracts and quickly play with Aptos features on-chain.
```
Comment (gregnazario) [inline apps/docusaurus/docs/tools/aptos-cli/index.md:9]: The nested and is strange.  How about? 
```suggestion
The Aptos command line interface (CLI) is a tool to help you compile, test, and quickly play with Move contracts and other Aptos features on-chain.
```

**Review by gregnazario — COMMENTED** [review]
Looks pretty good, I have some grammar feedback mostly

File: apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-windows.md (hunk 2 above)
Comment (jmintuitive) [reply apps/docusaurus/docs/tools/aptos-cli/install-cli/install-cli-windows.md:7]: Agreed :)

**Review by gregnazario — APPROVED** [review]

File: apps/docusaurus/docs/tools/aptos-cli/index.md (hunk 5 above)
Comment (jmintuitive) [reply apps/docusaurus/docs/tools/aptos-cli/index.md:9]: Separated into 2 sentences :)

**Pull request merged**